stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/WebSocketUtil.java

```diff
@@ -169,60 +178,168 @@ public static String getErrorMessage(Throwable err) {
+     */
+    public static boolean hasRetryConfig(ObjectValue webSocketClient) {
+        return webSocketClient.getMapValue(CLIENT_ENDPOINT_CONFIG).
+                getMapValue(WebSocketConstants.RETRY_CONFIG) != null;
     }
 
-    public static MapValue<String, Object> createDetailRecord(String errMsg, ErrorValue cause) {
-        MapValue<String, Object> detail = BallerinaValues.createRecordValue(HttpConstants.PROTOCOL_HTTP_PKG_ID,
-                                                                            WebSocketConstants.WEBSOCKET_ERROR_DETAILS);
-        return BallerinaValues.createRecord(detail, errMsg, cause);
+    private static void waitForHandshake(ObjectValue webSocketClient, CountDownLatch countDownLatch) {
+
+        long timeout = WebSocketUtil.findTimeoutInSeconds((MapValue<String, Object>) webSocketClient.getMapValue(
+                HttpConstants.CLIENT_ENDPOINT_CONFIG), "handShakeTimeoutInSeconds", 300);
+        try {
+            if (!countDownLatch.await(timeout, TimeUnit.SECONDS)) {
+                throw new WebSocketException(WebSocketConstants.ErrorCode.WsGenericError,
+                        "Waiting for WebSocket handshake has not been successful", WebSocketUtil.createErrorCause(
+                        "Connection timeout", IOConstants.ErrorCode.ConnectionTimedOut.errorCode(),
+                        IOConstants.IO_PACKAGE_ID));
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new WebSocketException(WebSocketConstants.ERROR_MESSAGE + e.getMessage());
+        }
+    }
+
+    /**
+     * Reconnect when the WebSocket connection is lost.
+     *
+     * @param connectionInfo - information about the connection.
+     * @return If attempts reconnection, then return true.
+     */
+    public static boolean reconnect(WebSocketConnectionInfo connectionInfo) {
+        ObjectValue webSocketClient = connectionInfo.getWebSocketEndpoint();
+        RetryContext retryConnectorConfig = (RetryContext) webSocketClient.getNativeData(WebSocketConstants.
+                RETRY_CONFIG);
+        int interval = retryConnectorConfig.getInterval();
+        int maxInterval = retryConnectorConfig.getMaxInterval();
+        int maxAttempts = retryConnectorConfig.getMaxAttempts();
+        int noOfReconnectAttempts = retryConnectorConfig.getReconnectAttempts();
+        float backOfFactor = retryConnectorConfig.getBackOfFactor();
+        WebSocketService wsService = connectionInfo.getService();
+        Date date = new Date();
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
+        // Check if the no of reconnection attempts is equal to the maximum number of reconnection attempts or not.
+        // If it is not equal, then call the `initialiseWebSocketConnection()` function.
+        // If it is equal, then return false.
+        if (((noOfReconnectAttempts < maxAttempts) && maxAttempts > 0) || maxAttempts == 0) {
```

Comment:
Why do we need to check if `maxAttempts>0` ? When will it become negative?

Reply:
Negative case handles at https://github.com/ballerina-platform/ballerina-lang/pull/19723/files#diff-89edaaa1baefeea805366138354f0258R225.

If maxAttempts = 0,  we need to retry infinitely.  so check this condition here.

Reply:
That means when it comes to this location it will not be negative right? Do we need the check?

Reply:
Removed it